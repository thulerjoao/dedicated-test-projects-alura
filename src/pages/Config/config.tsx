import Formulario from "../../components/form/form";
import List from "../../components/list/list";
import Rodape from "../../components/rodape/rodape";

const Config = () => {
  return (
    <>
      <Formulario />
      <List />
      <Rodape />
    </>
  );
};

export default Config;
