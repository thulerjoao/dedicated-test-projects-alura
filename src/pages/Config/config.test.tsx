import { render } from "@testing-library/react";
import { RecoilRoot } from "recoil";
import Config from "./config";

const mockNavigation = jest.fn();

jest.mock("react-router-dom", () => {
  return {
    useNavigate: () => mockNavigation,
  };
});
describe("pagina de configuraÃ§ao", () => {
  test("deve ser renderizada corretamente", () => {
    const { container } = render(
      <RecoilRoot>
        <Config />
      </RecoilRoot>
    );
    expect(container).toMatchSnapshot();
  });
});
