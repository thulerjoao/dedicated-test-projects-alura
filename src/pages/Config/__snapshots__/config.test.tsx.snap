// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pagina de configura√ßao deve ser renderizada corretamente 1`] = `
<div>
  <form>
    <input
      placeholder="Insira os nomes dos participantes"
      value=""
    />
    <button
      disabled=""
    >
      Adicionar
    </button>
  </form>
  <ul />
  <footer>
    <button
      disabled=""
    >
      Iniciar Brincadeira
    </button>
  </footer>
</div>
`;
